### PR TITLE
Split the exl2 weight ASAP.

### PR DESCRIPTION
**Pull request by sgsdxzy**
Reduce vram usage during exl2 model loading.

### CODE REVIEW

**Comment by sgsdxzy** [comment]
The outer `model.named_parameters()` loop during loading still keeps a reference to weights so we have to force release memory by setting `.data=torch.empty(0, device="cpu")`.
`q_invperm` is not used later in the kernel, probably we should remove it too after loading.

**Comment by sgsdxzy** [comment]
Sharding on CPU then copying the contiguous tensor reduces GPU vram bubble, increasing GPU blocks further (36k->41k ctx for me).

File: aphrodite/modeling/layers/linear.py [hunk 1]
```diff
@@ -26,6 +26,69 @@ def adjust_marlin_shard(param, shard_size, shard_offset):
     return shard_size * marlin_tile_size, shard_offset * marlin_tile_size
 
 
+# Split the exl2 weight at group boundary
```
Comment (AlpinDale) [inline aphrodite/modeling/layers/linear.py:29]: Why is this moved to this file?

Comment (sgsdxzy) [reply aphrodite/modeling/layers/linear.py:29]: It would be a circular import otherwise.

**Review by AlpinDale — APPROVED** [review]
Thanks for the PR! This should fix SQ+ as well.
